test/e2e/resource_placement_ro_test.go

```diff
@@ -0,0 +1,1134 @@
+			wantRONames := []placementv1beta1.NamespacedName{
+				{Namespace: workNamespace, Name: fmt.Sprintf(placementv1beta1.OverrideSnapshotNameFmt, roName, 0)},
+			}
+			rpStatusUpdatedActual := rpStatusWithOverrideUpdatedActual(appConfigMapIdentifiers(), allMemberClusterNames, "0", nil, wantRONames)
+			Eventually(rpStatusUpdatedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update RP %s status as expected", rpName)
+		})
+
+		// This check will ignore the annotation of resources.
+		It("should place the resources on all member clusters", checkIfPlacedWorkResourcesOnAllMemberClusters)
+
+		It("should have override annotations on the configmap", func() {
+			want := map[string]string{roTestAnnotationKey: roTestAnnotationValue}
+			checkIfOverrideAnnotationsOnAllMemberClusters(false, want)
+		})
+
+		It("update ro and change annotation value", func() {
+			Eventually(func() error {
+				ro := &placementv1beta1.ResourceOverride{}
+				if err := hubClient.Get(ctx, types.NamespacedName{Name: roName, Namespace: workNamespace}, ro); err != nil {
+					return err
+				}
+				ro.Spec = placementv1beta1.ResourceOverrideSpec{
+					Placement: &placementv1beta1.PlacementRef{
+						Name:  rpName, // assigned RP name
+						Scope: placementv1beta1.NamespaceScoped,
+					},
+					ResourceSelectors: configMapOverrideSelector(),
+					Policy: &placementv1beta1.OverridePolicy{
+						OverrideRules: []placementv1beta1.OverrideRule{
+							{
+								ClusterSelector: &placementv1beta1.ClusterSelector{
+									ClusterSelectorTerms: []placementv1beta1.ClusterSelectorTerm{},
+								},
+								JSONPatchOverrides: []placementv1beta1.JSONPatchOverride{
+									{
+										Operator: placementv1beta1.JSONPatchOverrideOpAdd,
+										Path:     "/metadata/annotations",
+										Value:    apiextensionsv1.JSON{Raw: []byte(fmt.Sprintf(`{"%s": "%s"}`, roTestAnnotationKey, roTestAnnotationValue1))},
+									},
+								},
+							},
+						},
+					},
+				}
+				return hubClient.Update(ctx, ro)
+			}, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update ro as expected", rpName)
+		})
+
+		It("should update RP status as expected after RO update", func() {
+			wantRONames := []placementv1beta1.NamespacedName{
+				{Namespace: workNamespace, Name: fmt.Sprintf(placementv1beta1.OverrideSnapshotNameFmt, roName, 1)},
+			}
+			rpStatusUpdatedActual := rpStatusWithOverrideUpdatedActual(appConfigMapIdentifiers(), allMemberClusterNames, "0", nil, wantRONames)
+			Eventually(rpStatusUpdatedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update RP %s status as expected", rpName)
+		})
+
+		// This check will ignore the annotation of resources.
+		It("should place the selected resources on member clusters after RO update", checkIfPlacedWorkResourcesOnAllMemberClusters)
+
+		It("should have updated override annotations on the configmap", func() {
```

Comment:
missing the test to update the ro and no updates on the configmap itself? similar to CRP

Reply:
AI must missed this part. Added